tests/test_testing.py

```diff
@@ -293,11 +298,18 @@ def setUpClass(cls):
         super().setUpClass()
         assert urlpatterns is cls.urlpatterns
 
-    @classmethod
-    def tearDownClass(cls):
-        assert urlpatterns is cls.urlpatterns
-        super().tearDownClass()
-        assert urlpatterns is not cls.urlpatterns
+        if django.VERSION > (4, 0):
+            cls.addClassCleanup(
+                check_urlpatterns,
+                cls
+            )
+
+    if django.VERSION < (4, 0):
+        @classmethod
+        def tearDownClass(cls):
+            assert urlpatterns is cls.urlpatterns
+            super().tearDownClass()
+            assert urlpatterns is not cls.urlpatterns
```

Comment:
Same `tearDownClass` vs `addClassCleanup` question as above.

Nothing about it in the release notes... https://docs.djangoproject.com/en/4.0/releases/4.0/ so I'm curious what prompts this to need changing?

Reply:
See the commit I linked — we no longer call `tearDown` ourselves so `super()` isn't doing what it used to. (I presume the base implementation is empty.) 

I didn't trace the exact call flow... — timings on those enable/disable or `__enter__`/`__exit__` s is tricky, but there were failures without following the change in Django.

Reply:
Hrm that's interesting yeah. Looking through the commit in django/django@faba5b7 I can't *see* any reason why subclasses would need to stop using `tearDownClass`, even if Django 4.0 isn't using it directly for it's own implementation. But there's possibly some change in interaction I've not figured out there.

Reply:
Yeah, no 😀 — it's not obvious. 

My suspicion it's the timing on [the `override_settings` usage](https://github.com/encode/django-rest-framework/blob/f0a5b958a134e8cd94e3ef3263e8fa623ac9b82f/rest_framework/test.py#L380) — which is always _delicate_, as we know from e.g. #5668 &friends.

Reply:
Once `super().setUpClass()` is moved in `rest_framework/test.py` this class doesn't need to change at all.
(Or at least, that's how it seems from my testing.)

Reply:
> My suspicion it's the timing on the override_settings usage — which is always delicate, as we know from e.g. #5668 &friends.

Yes that seems *very* likely.

Reply:
Okay, if you want to adjust please do! (Otherwise I'll have another look later on)